### DISCUSSION TITLE
How to add an external force on a certain point of the platform?

### DISCUSSION THREAD

**Discussion by YuYuJou** [OpenFAST/openfast#2471] [Q&A]
I want to get the force_displacement curve, but I don't know how to add the external force, or there is an output of restoring force?
Thanks

**Comment by jjonkman** [comment]
Dear @YuYuJou,

Presumably you are referring to the force-displacement curve of the mooring system of a floating offshore wind turbine.  To generate this, you can output the reaction forces/tension at the fairlead from the mooring module you are using, related to the platform displacement.

To add user-prescribed external point loads (forces, moments) to an OpenFAST model, you can use the `StC_DOF_MODE` = 4 option of the Structural Control (StC) submodel of ServoDyn, with one point load implemented per instance of StC in the blades, nacelle, tower, or substructure.  See the ServoDyn (https://openfast.readthedocs.io/en/main/source/user/servodyn/input.html#servodyn-primary-input-file) and StC (https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_input.html) input file documentation on readthe docs for more information.

Best regards,

**Reply by YuYuJou** [reply]
Sorry, I tried and read the example file.
But it didn't work.
![image](https://github.com/user-attachments/assets/1d5256f1-a725-4593-94d6-251886ac5873)
![image](https://github.com/user-attachments/assets/4604d773-1171-4e95-995f-9ff3b2391943)

**Reply by jjonkman** [reply]
ServoDyn input `SStCfiles` should point to the name of an StC input file, not a prescribed load time series file.  The name of the prescribed load time series file should be specified within the StC input file.  And presumably, you meant to set `StC_DOF_MODE` to 4 in the StC file rather than `NumSStC` to 4.  (Setting `NumSStC` to 4 would generate 4 StCs in the substructure, which I presume is not what you want to do.)

Best regards,

**Comment by YuYuJou** [comment]
Is it correct that I create a new file to write StC file?
![image](https://github.com/user-attachments/assets/6a304c67-4684-423c-bb22-2b8428b86e01)
![image](https://github.com/user-attachments/assets/52b674ea-bd72-47cd-a6d2-8a551013fd8b)
![image](https://github.com/user-attachments/assets/6a729b5a-046a-4b28-bfd3-effd0d6cece7)
![image](https://github.com/user-attachments/assets/ed654291-23e6-40a5-8d6c-188f243dc764)
I don't understand what [The variable "Echo"] means

**Reply by jjonkman** [reply]
Dear @YuYuJou,

Like other OpenFAST input files, the StC file requires a specific format.  The StC input file is documented within OpenFAST readthedocs here: https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_input.html.  And you can find an example StC input file in the r-test: https://github.com/OpenFAST/r-test/blob/main/glue-codes/openfast/5MW_Baseline/NRELOffshrBsline5MW_ServoDyn_StC.dat.  `Echo` is the first input on line 4 of the StC input file.

Best rears,
